DataMigration/src/main/java/org/sagebionetworks/dian/datamigration/tools/schedulev2/ScheduleV2Migration.java

```diff
@@ -196,16 +196,48 @@ public static TestSchedule createHMSchedule(String uId, String scheduleJson) thr
     }
 
     public static void
-    createV2Schedule(SageV1Schedule v1Schedule, String uId, String sId) throws IOException {
+    createV2Schedule(String arcId, SageV1Schedule v1Schedule, String uId, String sId) throws IOException {
 
         String iANATimezone = getTimezone(v1Schedule);
+        System.out.println("User timezone is " + iANATimezone);
+
+        DateTime studyStart = SageScheduleController.Companion
+                .createDateTime(v1Schedule.getStudyBursts().get(0).getStartDate(), iANATimezone)
+                // Putting this date at noon, gives Bridge the most flexibility with
+                // how it adjusts local times and time zones to get the correct start day
+                .withTimeAtStartOfDay().plusHours(12);
+
         for (int i = 0; i < v1Schedule.getStudyBursts().size(); i++) {
             SageV1StudyBurst studyBurst = v1Schedule.getStudyBursts().get(i);
 
             // The new V2 Sage scheduling does not include the practice test, and so
             // the study as a whole, and the first study burst schedule starts.
             DateTime startDate = SageScheduleController.Companion
-                    .createDateTime(studyBurst.getStartDate());
+                    .createDateTime(studyBurst.getStartDate(), iANATimezone)
+                    // Putting this date at 1 PM, gives Bridge the most flexibility with
+                    // how it adjusts local times and time zones to get the correct start day
+                    // Add one additional hour to account for any daylight savings issues
+                    .withTimeAtStartOfDay().plusHours(13);
+
+            // HappyMedium's app had future study bursts that were off by 1 day and these
+            // make the dashboard looks disjointed.
+            // This doesn't happen all the time, so first check for the situation.
+            // To fix it, move each future study burst start date by 1 day
+            if (startDate.isAfterNow()) {
+                // plus 1 hour accounts for daylight savings time issues
+                int daysFromStart = Days.daysBetween(studyStart, startDate).getDays();
+                int expectedDays = SageScheduleController.Companion.getWeeksBetweenStudyBursts() *
+                        SageScheduleController.Companion.getDaysInAllStudyBursts() * i;
```

Comment:
Would this work if getDaysInAllStudyBursts() returned a value that was not 7? I see the first study burst is 8, but i is 0 in that case. Could it just be 7 * i ?

Reply:
The business logic that the first study burst has 8 days has actually been removed and no longer applies. All study bursts have 7 days in the new app.